pkg/azure/azure.go

```diff
@@ -38,19 +38,22 @@ func NewCloud(info *CloudInfo) api.Cloud {
 }
 
 func (az *azureCloud) PrepareForSubmariner(input api.PrepareForSubmarinerInput, reporter reporterInterface.Interface) error {
+	return az.OpenPorts(input.InternalPorts, reporter)
+}
+
+func (az *azureCloud) OpenPorts(ports []api.PortSpec, reporter reporterInterface.Interface) error {
 	reporter.Start("Opening internal ports for intra-cluster communications on Azure")
```

Comment:
Even with the API simplification, this should still avoid causing errors when called with no ports:

```suggestion
	if (len(ports) == 0) {
		return nil
	}
	
	reporter.Start("Opening internal ports for intra-cluster communications on Azure")
```

Reply:
IDK if I support this position, would you call a command that does something with not enough arguments and expect it to silently succeed? Not usually, unless you use some`-f` flag.

If anything, this should return an error.

Reply:
I agree with @skitt. It's better to handle the trivial case here rather than forcing all users to do it or risk a latent bug.

Reply:
Why would it be acceptable in this case to send an empty list? It should be an error

Reply:
Somebody has to verify/handle an empty list - it might as well be at the source rather than expecting all callers to check for it - makes it a simpler API to use. I don't consider it an error - the API is to open the given ports - the trivial case is that the list happens to be empty in which case it's a noop. There's many similar trivial cases in other APIs, eg a sort function where it simply returns if the input list is empty instead of crashing or throwing an error.

Reply:
Regardless of the consensus here, its not part of this change, I'll send a separate PR with a proposal for this specifically.

Reply:
Please see #580